src/plugins/explore/public/application/pages/traces/trace_details/trace_view.tsx

```diff
@@ -0,0 +1,305 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React, { useEffect, useState, useMemo, useRef, useCallback } from 'react';
+import { i18n } from '@osd/i18n';
+import { EuiSpacer, EuiPanel, EuiLoadingSpinner, EuiResizableContainer } from '@elastic/eui';
+import './trace_view.scss';
+import { TraceTopNavMenu } from './public/top_nav_buttons';
+import { useOpenSearchDashboards } from '../../../../../../opensearch_dashboards_react/public';
+import { TracePPLService } from './server/ppl_request_trace';
+import { MountPoint } from '../../../../../../../core/public';
+import { transformPPLDataToTraceHits } from './public/traces/ppl_to_trace_hits';
+import { DataExplorerServices } from '../../../../../../data_explorer/public';
+import { generateColorMap } from './public/traces/generate_color_map';
+import { SpanDetailPanel } from './public/traces/span_detail_panel';
+import { ServiceMap } from './public/services/service_map';
+import { NoMatchMessage } from './public/utils/helper_functions';
+import { createTraceAppState } from './state/trace_app_state';
+import { SpanDetailSidebar } from './public/traces/span_detail_sidebar';
+
+export interface TraceDetailsProps {
+  setMenuMountPoint?: (mount: MountPoint | undefined) => void;
+}
+
+export const TraceDetails: React.FC<TraceDetailsProps> = ({ setMenuMountPoint }) => {
+  const {
+    services: { chrome, data, osdUrlStateStorage },
+  } = useOpenSearchDashboards<DataExplorerServices>();
+
+  // Initialize URL state management
+  const { stateContainer, stopStateSync } = useMemo(() => {
+    return createTraceAppState({
+      stateDefaults: {
+        traceId: '',
+        dataSourceId: '',
+        indexPattern: 'otel-v1-apm-span-*',
```

Comment:
Do we have to hardcode for indexPattern? Also can we use dataset as in explore as indexPattern has been migrated?

Reply:
The re-direction should provide the index pattern in the url, this is just a fallback to the default naming convention if it is not provided.

I haven't seen the query example version using the dataset instead of index pattern. As soon as I see a ppl example of what the call looks like that should be simple to switch out.